### DISCUSSION TITLE
Is it possible to change the highlighting of some specific text?

### DISCUSSION THREAD

**Discussion by mark-summerfield** [rouge-ruby/rouge#2025] [Q&A]
I am using asciidoctor-pdf with the rouge formatter.
I have defined a custom theme based on the rouge pastie theme. Here's my `custom.rb` file:
```
require 'rouge' unless defined? ::Rouge.version

module Rouge; module Themes
  class Custom < Pastie
    name 'custom'

    style Comment,         :fg => '#666666', :italic => true
    style Str,             :fg => '#808000'
    style Str::Escape,     :fg => '#808000'
    style Str::Interpol,   :fg => '#808000'
    style Str::Other,      :fg => '#808000'
    style Name::Constant,  :fg => '#000080'
    style Keyword,         :fg => '#000080', :bold => true
    style Keyword::Type,   :fg => '#80604D', :bold => true
    style Operator::Word,  :fg => '#008800', :bold => true
    style Num,             :fg => '#B400B4'
  end
end; end
```
I would like to make one addition. I want to color a specific word in the highlighted text ("nil"). This word is actually a `Name::Constant`; so I need to override that setting for this one word.
Is this possible?

**Comment by i-koch** [comment]
You have to add the appropriate rule for "nil" to the lexer which is used to colorize your language.
To be sure that it is included in future releases of rouge, you should fork and issue a pull request.
But be aware that the different implementations of asciidoctor might ship their own rouge and might not be updated on a regular base, e.g. AsciidoctorFX.

**Comment by jneen** [comment]
You might consider using the `escape` lexer to insert arbitrary HTML given certain delimiters.

If you use e.g. `escape?lang=asciidoctor-pdf` as your language selection, then you can use `<! ... !>` to generate Escape tokens, which are arbitrary output that will be included verbatim, unescaped by the formatter. Great for blogs or documents, but don't use this for untrusted code - and for safety purposes the standard formatters will ignore Escape tokens unless `Rouge::Formatter.enable_escape!` has been called.

**Comment by mark-summerfield** [comment]
Ok thanks for the help.
